workspaces/src/network/account.rs

```diff
@@ -40,7 +101,188 @@ impl Contract {
+    /// yields a JSON string object.
+    pub async fn view<T: Network>(
+        &self,
+        worker: &Worker<T>,
+        function: String,
+        args: Vec<u8>,
+    ) -> anyhow::Result<String> {
+        worker.view(self.id().clone(), function, args).await
+    }
+
+    /// Deletes the current contract, and returns the execution details of this
+    /// transaction. The beneciary will receive the funds of the account deleted
+    pub async fn delete_contract<T: Network>(
+        self,
+        worker: &Worker<T>,
+        beneficiary_id: AccountId,
+    ) -> anyhow::Result<CallExecutionDetails> {
+        self.account.delete_account(worker, beneficiary_id).await
+    }
+}
+
+pub struct CallBuilder<'a, T> {
+    worker: &'a Worker<T>,
+    signer: InMemorySigner,
+    contract_id: AccountId,
+
+    function: String,
+    args: Option<Vec<u8>>,
+    deposit: Option<u128>,
+    gas: Option<u64>,
+}
+
+impl<'a, T: Network> CallBuilder<'a, T> {
+    fn new(
+        worker: &'a Worker<T>,
+        contract_id: AccountId,
+        signer: InMemorySigner,
+        function: String,
+    ) -> Self {
+        Self {
+            worker,
+            signer,
+            contract_id,
+            function,
+            args: None,
+            deposit: None,
+            gas: None,
+        }
+    }
+
+    pub fn with_args(mut self, args: Vec<u8>) -> Self {
+        self.args = Some(args);
+        self
+    }
+
+    pub fn with_deposit(mut self, deposit: u128) -> Self {
+        self.deposit = Some(deposit);
+        self
+    }
+
+    pub fn with_gas(mut self, gas: u64) -> Self {
+        self.gas = Some(gas);
+        self
+    }
+
+    pub async fn transact(self) -> anyhow::Result<CallExecutionDetails> {
+        self.worker
+            .client()
+            .call(
+                &self.signer,
+                self.contract_id,
+                self.function,
+                self.args
+                    .expect("required `with_args` to be specified as apart of Call"),
+                self.gas,
+                self.deposit,
+            )
+            .await
+            .map(Into::into)
+    }
+
+    pub async fn view(self) -> anyhow::Result<Vec<u8>> {
+        self.worker
+            .client()
+            .view(
+                self.contract_id,
+                self.function,
+                self.args
+                    .expect("required `with_args` to be specified as apart of View"),
+            )
+            .await
+            .map(Into::into)
+    }
+}
+
+pub struct CreateAccountBuilder<'a, T> {
+    worker: &'a Worker<T>,
+    signer: InMemorySigner,
+    parent_id: AccountId,
+    new_account_id: String,
+
+    initial_balance: Balance,
+    secret_key: Option<SecretKey>,
+}
+
+impl<'a, T> CreateAccountBuilder<'a, T>
+where
+    T: Network,
+{
+    fn new(
+        worker: &'a Worker<T>,
+        signer: InMemorySigner,
+        parent_id: AccountId,
+        new_account_id: String,
```

Comment:
I'm a little confused on why we are forcing the account to be built as a subaccount. If this is specifically intended for that flow, perhaps we should rename to indicate this?

Reply:
I'll just rename it to be `create_subaccount` then. This was initially to be consistent with `workspaces-js/Account`, but better to be explicit here

Reply:
is the fact that it's a subaccount obfuscated completely from the API? The only issue I see is just that it may be doing something differently than a developer may expect, which is a bit of a foot gun. Maybe we should consider renaming on js side for consistency as well if we want to make this change?

Reply:
> is the fact that it's a subaccount obfuscated completely from the API?

Yup, it's pretty much that. When we go to call into account creation, there's no flag to say, "hey let's go make a subaccount". Instead, it's just based on account id we give it such as "A" or "B.A"

> Maybe we should consider renaming on js side for consistency as well if we want to make this change?

The js side is pretty flexible (maybe too flexible). The `create_account` on that side allows you to create both top level and subaccounts regardless of what `account` context is present. Changing it on the js side would probably require a good amount of rework. On rust side, I've just had it be explicit in the functionality. If you want to create a TLA account you call `worker.create_tla`, and subaccounts with `account.create_subaccount`

Reply:
okay, I think I might have to look through code when I get a chance to understand this difference, thanks for giving context!